### PR TITLE
feat: adds hvd landing data fetching

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-kshvd-data-fetching-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1204908934882872/1205890213689119) 🎟️

## 🗒️ What

Sets up data fetching and basic scaffold of the hvd landing page. I created spaces in the `views` directory for both the landing and guide routes. I also set up the `BaseLayout` for both routes. Also upgrades `@types/node` to align with `v18`. 

## 🧪 Testing

- Visit the [hvd landing page](https://dev-portal-git-kshvd-data-fetching-hashicorp.vercel.app/validated-designs), validate that the data from the repo renders and it links to the guide

## 💭 Anything else?

Copies over a lot of groundwork laid by @zchsh in https://github.com/hashicorp/dev-portal-internal/pull/131

### CODE REVIEW

File: src/views/homepage/components/featured-content-grid/search-featured-card.tsx [hunk 1]
```diff
@@ -140,7 +140,7 @@ const SearchFeaturedCard = () => {
 		)
 
 		// If auto scroll is enabeld, set up the animation interval
-		let interval: NodeJS.Timer
+		let interval: NodeJS.Timeout
```
Comment (kendallstrautman) [inline src/views/homepage/components/featured-content-grid/search-featured-card.tsx:143]: Little type fix after updating @types/node

File: .github/workflows/nextjs_bundle_analysis.yml [hunk 2]
```diff
@@ -44,6 +44,7 @@ jobs:
       - name: Build next.js app
         env:
           SKIP_BUILD_PRODUCT_REDIRECTS: 1
+          GITHUB_TOKEN: ${{ secrets.CI_GH_TOKEN }}
```
Comment (kendallstrautman) [inline .github/workflows/nextjs_bundle_analysis.yml:47]: Had to add this otherwise the repo `404`d

File: scripts/extract-hvd-content.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import fs from 'fs'
```
Comment (kendallstrautman) [inline scripts/extract-hvd-content.ts:1]: Based on https://github.com/hashicorp/dev-portal-internal/pull/131

File: src/lib/fetch-github-archive-zip.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { Octokit } from '@octokit/core'
```
Comment (kendallstrautman) [inline src/lib/fetch-github-archive-zip.ts:1]: Copied directly from https://github.com/hashicorp/dev-portal-internal/pull/131

File: src/pages/validated-designs/[...slug]/index.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import ValidatedDesignGuideView from 'views/validated-designs/guide'
+
+// @TODO make sure to set it up so that the base `guideSlug` path redirects to the first section page
+// for example /validated-designs/terraform-operation-guides-adoption would render the content for
+// /validated-designs/terraform-operation-guides-adoption/0000-introduction
+export async function getStaticPaths() {
+	// @TODO refactor to dynamically build up static paths here for all
+	// products/hvds/and hvd section pages based on contents of hvd-docs filesystem
+	return {
+		paths: [
```
Comment (kendallstrautman) [inline src/pages/validated-designs/[...slug]/index.tsx:15]: Just hard coding these paths for now, we'll wire up these routes in a [separate ticket ](https://app.asana.com/0/1204908934882872/1205942535702199)

File: src/pages/validated-designs/index.tsx [hunk 6]
```diff
@@ -3,26 +3,22 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import Head from 'next/head'
-
-export default function ValidatedDesignsLanding() {
```
Comment (kendallstrautman) [inline src/pages/validated-designs/index.tsx:8]: Moved this to the `views` dir.

File: src/views/validated-designs/guide/index.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+import Head from 'next/head'
+import BaseLayout from 'layouts/base-layout'
+import MobileMenuLevelsGeneric from 'components/mobile-menu-levels-generic'
+import InlineLink from 'components/inline-link'
+
+export default function ValidatedDesignGuideView() {
+	return (
+		<BaseLayout mobileMenuSlot={<MobileMenuLevelsGeneric />}>
```
Comment (kendallstrautman) [inline src/views/validated-designs/guide/index.tsx:13]: Added `BaseLayout` here so it doesn't look totally odd.
Screenshot 2023-11-13 at 4 50 42 PM

File: src/views/validated-designs/index.tsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+	product: ProductSlug
+	guides: HvdGuide[]
+}
+
+interface HvdGuide {
+	slug: string
+	title: string
+	description: string
+	href: string
+}
```
Comment (kendallstrautman) [inline src/views/validated-designs/index.tsx:34]: Definitely not married to these types, but they were my interpretation of the view's needs based on the current designs.

Comment (RubenSandwich) [reply src/views/validated-designs/index.tsx:34]: The only thing that kind of pops into my head here is to also add the file path for the HvdGuide and HvdCategoryGroup, but I'm unsure how useful that would be currently. I'm gonna need to play with this PR a bit to figure out what changes I feel like are needed. But I do think this is a great start and follows a "List -> Detail" data pattern.

**Review by RubenSandwich — APPROVED** [review]
Thanks so much! 🚀

**Pull request merged**